### DISCUSSION TITLE
don't support Unicode?

### DISCUSSION THREAD

**Discussion by KamiyaMinoru** [sumatrapdfreader/sumatrapdf#3563] [Q&A]
I tried to copy (Chinese) text in a PDF but get garbled result

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/78710607/3c48979d-ec48-4898-a5ca-5c29883f8b93)

**Comment by GitHubRulesOK** [comment]
if its aPDF its usually not Unicode unless using future PDF format most PDF use legacy PDF based ansi or UTF16 as hex or any other encoding but never UTF-8 thus windows copy and paste as UTF-8 is rarely good for non Plain American Character text.

It depends what and how it was stored, as to how well it extracts in any application

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/6f3c7ef0-ef4d-4159-a484-f53e25547c0d)

**Reply by KamiyaMinoru** [reply]
I can't take any action to modify the PDF I got; in this case how do I figure it out?
file address: https://cultureexpress.taipei/Publication/C000007

**Reply by GitHubRulesOK** [reply]
too many slow links to China ?
which document that page is on would help but none are loading well (keep dropping out ) perhaps see if you can upload to say https://www.DocDroid.net but do not know their size limit

- those files are huge with so many fonts they can only be listed on my screen from A-F the files have been exotically compressed so are very slow to work but the fonts seem generally well behaved.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/be826b44-8517-4b18-b279-d080d02f6bc1)

**Comment by GitHubRulesOK** [comment]
OK I found it (Object 2,148) but you can see its not human text
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/5f339f0a-008d-4f6a-b3d6-cdf7158a1e83)

the problem is that any extractor can only see something like this where that font is /C0_3 but here as 1 em high it makes no sense at all note this is not that exact string but a 6 character one where AA 36 9 4 68 E6 89 could be any thing at all English, Martian or possibly Chinese
/C0_3 1 Tf
0 -1.556 Td
< AA369468E689 >

**Reply by KamiyaMinoru** [reply]
Thanks! So I think it's problem of this PDF, and nothing to do in (any) reader.

**Reply by GitHubRulesOK** [reply]
As a Printout it is not a wrong PDF just no use for a blind person since those sounds do not look like normal words, PDF is designed for visual print out not for extraction of the components.

**Comment by dungsaga** [comment]
in this case it's better to use OCR to recognize the text

**Reply by GitHubRulesOK** [reply]
OCR is of limited value since it can only "Gusse" the single letters based on a Dictionary of known words. Hence usually abysmal for strings of numbers (it sees as random digits) or  here Chinese (Eastern languages such as Arabic around the world to Japanese) unless the dictionary looks as pairs of ordered characters like  word groups. It is always best to try to decode the binary  so here those 3 "letters" are in order `AA36 9468 E689` but that needs a dictionary of numbers to decode into characters which "should" be inside the PDF but often corrupted..

**Reply by dungsaga** [reply]
Some OCR tools claim to support Chinese (such as https://ocr.space/). I haven't tried that myself, but maybe it can help the OP.

**Reply by GitHubRulesOK** [reply]
@dungsaga you can try with that section (most of the others will not be a problem in SumatraPDF, but I forget which page "Haunt" is on but the file should be 7th JULY https://cultureexpress.taipei/PublicationEbookViewer/C000007/de559bef-1117-4822-99c1-fe8b5eb7dcbe
